### PR TITLE
feat: Set explicit resources on all containers

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/394

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: tests/templates/kuttl/smoke/00-limit-range.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: limit-request-ratio
+spec:
+  limits:
+  - type: "Container"
+    maxLimitRequestRatio:
+      cpu: 5
```
Comment (adwk67) [inline tests/templates/kuttl/smoke/00-limit-range.yaml:10]: Running this tests results in the following error:

> create Pod hello-world-server-default-0 in StatefulSet hello-world-server-default failed error: pods "hello-world-server-default-0" is forbidden: cpu max limit to request ratio per Container is 5, but provided ratio is 20.000000

Changing this to `cpu: 20` results in a successful test.

Comment (sbernauer) [reply tests/templates/kuttl/smoke/00-limit-range.yaml:10]: Weird, tests passed on my machine...

Comment (sbernauer) [reply tests/templates/kuttl/smoke/00-limit-range.yaml:10]: Nevermind, found the problem

**Comment by sbernauer** [comment]
```
--- PASS: kuttl (35.91s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke (34.32s)
        --- PASS: kuttl/harness/check-name (35.88s)
```

**Comment by sbernauer** [comment]
bors r+
Thanks for quick review!

**Pull request merged**